core/src/main/java/software/amazon/awssdk/util/DateUtils.java

```diff
@@ -65,82 +93,31 @@
-                // Normal case: nothing special here
-                return new Date(ISO_8601_DATE_FORMAT.parseMillis(dateString));
-            }
-            // Handling edge case:
-            // Joda-time can only handle up to year 292278993 but we are given
-            // 292278994;  So we parse the date string by first adjusting
-            // the year to 292278993. Then we add 1 year back afterwards.
-            final long milliLess365Days = ISO_8601_DATE_FORMAT.parseMillis(temp);
-            final long milli = milliLess365Days + MILLI_SECONDS_OF_365_DAYS;
-            if (milli < 0) { // overflow!
-                // re-parse the original date string using JodaTime so as to
-                // throw  an exception with a consistent message
-                return new Date(ISO_8601_DATE_FORMAT.parseMillis(dateString));
-            }
-            return new Date(milli);
-        } catch (IllegalArgumentException e) {
-            try {
-                return new Date(ALTERNATE_ISO_8601_DATE_FORMAT.parseMillis(dateString));
-                // If the first ISO 8601 parser didn't work, try the alternate
-                // version which doesn't include fractional seconds
-            } catch (Exception oops) {
-                // no the alternative route doesn't work; let's bubble up the original exception
-                throw e;
-            }
+            millis = parseIso8601DateAsMillis(dateString, ISO_8601_DATETIME_FORMAT);
+        } catch (DateTimeParseException e) {
+            millis = parseIso8601DateAsMillis(dateString, ALTERNATE_ISO_8601_DATE_FORMAT);
         }
+        return new Date(millis);
     }
 
-    /**
-     * Returns a date string with the prefix temporarily substituted, if
-     * applicable, so that JodaTime can handle it.  Otherwise, if not applicable,
-     * the original date string is returned.
-     * <p>
-     * See https://github.com/aws/aws-sdk-java/issues/233
-     */
-    private static String tempDateStringForJodaTime(String dateString) {
-        final String fromPrefix = "292278994-";
-        final String toPrefix = "292278993-";
-        return dateString.startsWith(fromPrefix)
-               ? toPrefix + dateString.substring(fromPrefix.length())
-               : dateString;
-    }
-
-    /**
-     * Returns the original runtime exception iff the joda-time being used
-     * at runtime behaves as expected.
-     *
-     * @throws IllegalStateException if the joda-time being used at runtime
-     *     doens't appear to be of the right version.
-     */
-    private static <E extends RuntimeException> E handleException(E ex) {
-        if (JodaTime.hasExpectedBehavior()) {
-            return ex;
+    private static long parseIso8601DateAsMillis(final String dateString, DateTimeFormatter formatter) {
+        // https://github.com/aws/aws-sdk-java/issues/233
+        // Handling edge case: java time required '+' sign in front of the year 292278994 (or greater).
```

Comment:
the new `java.time` classes shouldn't have a problem with this specific value anymore. In the original issue, the problem was that it was possible to serialize DDB items with date values equal to Long.MAX_VALUE but the retrieved items could not be deserialized. Since we've moved over to replacing `Date` with `Instant`, I think we need to be able to handle `Instant.MAX`